### ISSUE TITLE
Mr Robot S03e02 brought me here ;)

### ISSUE THREAD

**Issue by Dingo64** [RedBalloonShenanigans/MonitorDarkly#16]
Anybody from Mr Robot here?

**Comment by rinaldi-rahmanda** [comment]
same here lol

**Comment by 0xac** [comment]
I am shamefully behind on watching Mr. Robot. I will stop what I'm doing
now and go watch some TV -)

**Comment by Dingo64** [comment]
You really should, they've just made your Github popular! 

https://i.imgur.com/HCk7PId.png

**Comment by Spacefish** [comment]
+1 me too :)
but they might change the content on the server where the base64 string is placed!

b.t.w. jBouaqK9R8jXxfpE6kGV ;)

**Comment by gabriele92** [comment]
Here from Mr Robot decoding too! :) Loved this easter egg!

**Comment by Dingo64** [comment]
@Spacefish I noticed the filename too. Did you decode it already?

**Comment by Spacefish** [comment]
@Dingo64 no :(, tried different rotational ciphers (ROT1-25), base64 and googled it, but did not find anything.
It might just be a random generated string by the tool which generated the QR Code image.

It´s 20 characters and number to character ratio is pretty much normal for a random 62 symbol alphabet (26 lowercase 26 uppercase and 10 digits = 62 symbols)

thought about a hash value as well, it might be a hash for a DHT based bittorent download (magnet link). But there are no hashes which use 20 characters with a 62 symbol alphabet i know of..
SHA1 is typically 160bit and written in 32 characters which contain numbers 0-9 and letters A-Z (no lower uppercase differntiation).
MD5 is written in HEX typically
SHA256 or SHA2 is way too long..

The 20 character string is too long to be a youtube video identifier.

**Comment by Dingo64** [comment]
@Spacefish I tried base64, ROT13 and ROT47. I think it might be a password. Doesn't look like base64 encoded hash because I don't think there is a hash function that returns 120 bits.

**Comment by Spacefish** [comment]
@Dingo64 updated my comment, i tried different things as well, but i don´t think it´s a hash.. 
the symbol set does not allow to encode 6 bit per character as there are only 62 symbols, not 64! (26 lower characters 26 upper and 10 digits = 62)

if you analyse the frequency of the different symbols in jBouaqK9R8jXxfpE6kGV
each symbol is there only once, this hints at a random string as well, but the string is only 20 characters though!

**Comment by Dingo64** [comment]
@Spacefish You are right it can't be typical base 64 but there are variations of base 64 that are fine for filenames: https://en.wikipedia.org/wiki/Base64#Filenames 
So it can be base 62 but also base 64.

**Comment by Spacefish** [comment]
@Dingo64 yep that´s true! It might not contain any == padding and no _ or - as the data does not contain them and don´t need padding!
Maybe this whole string is just a login for: https://webmail.e-corp-usa.com/owa/index.php
as for example
Username: retro
Password: portal
downloads the pdf of elliots presentation 👍

**Comment by Dingo64** [comment]
@Spacefish When decoded as base62 it returns 20 E5 D1 5E 85 D5 3F A0 10 1A 58 3E 6A 2C 3C C7. This can be a 128 bit hash of something. But base62 is very rare and Elliot likes long passwords so perhaps just a password to something.

**Comment by Spacefish** [comment]
@Dingo64 
When Elliot logs into his PC counting the stars it´s a 21 character password.. the string we got is 20 characters long :(

**Comment by Dingo64** [comment]
@Spacefish If he logs with 21 chars to his PC he can log with 20 chars to something else.

**Comment by Spacefish** [comment]
B.t..w found this: http://www.e-corp-usa.com/users/072391/ it´s the QR Code on elliots employee card!

haha the adress is 135 E 57th Street, New York, NY 10022, right next to the Trump Tower b.t.w. lol :D

The creators of the series really pay attention to detail! Very nice!

**Comment by rtkd** [comment]
Now... everybody, back to solving Liber Primus.

**Comment by Spacefish** [comment]
@rtkd where is my book?
But my guess is, that the sequence has no real meaning and is just random.
The QR generator generates a random filename for the temp file, that's my guess!

**Comment by Dingo64** [comment]
@rtkd Can't now, decoding Voynich manuscript.

**Comment by rtkd** [comment]
@Spacefish You are literally only one click away.

**Comment by chrisbog94** [comment]
I'm here as well. InB4 4chan inb4 reddit

**Comment by fanick1** [comment]
wow, one never knows how deep the rabbit hole really goes

**Comment by evertonmj** [comment]
Whoa! I'm here because of MrRobot also :)

**Comment by MicTheSquid** [comment]
I am also here because of Mr. Robot.  Can anyone ELI5 what this file does?

**Comment by TteokbokkiNari** [comment]
Haha, fun to see so many people here!

**Comment by kirgy** [comment]
I thought they'd be more of us! I'm only just catching up on Prime, so I'm a bit late to the game. Soz guys.

**Comment by mwojt** [comment]
Big respect for the creators of the series

**Comment by Spacefish** [comment]
Probably a lot of "Nerds/Hipsters" watch the series without a real deep technical understanding of the details (well base64 isn´t that complicated of a thing).. But i guess this is the reason why we don´t see more people here.

While i was studying we had some of these people, they got all the latest "Nerd" gadgets but didn´t know what to do with them.. One guy got an Raspberry Pi a MakerBot and a Ocolus Rift. When asked what he did do with it:
- Well i installed Linux on the Pi (copied the standard image to an SD with a windows tool ^^)
- I only have the MakerBot for 3 weeks, i didn´t print anything yet
- There is a problem with the second monitor thing when i plug in the Rift and it didn´t work..

That´s like buying a car because it is cool, without owning a driver licence nor knowing how to use it..

**Comment by Dingo64** [comment]
@Spacefish You might be right. However I came here straight after watching the episode and decoding the link but as I noticed some people created a thread on Reddit instead. So maybe others just didn't have a need to post an issue here and thought Reddit is a better place for such discussions. 
I just hope @0xac isn't mad we are spamming his Github :) 

And a note to other fans of Mr. Robot- try not to create any other issues on this Github repo about Mr. Robot as this is very offtop and just post in this issue.

**Comment by annalisamf** [comment]
Here for Mr Robot too  😉

**Comment by 0xBADCODE** [comment]
Yup Mr Robot brought me here. Anyone work out whether jBouaqK9RjXxfpE6kGV is random?

**Comment by alessaba** [comment]
The satisfaction is very high 😍

**Comment by zinoharo** [comment]
what?

**Comment by alessaba** [comment]
Finding the egg

Filippo Claudi

**Comment by theclai** [comment]
Me too

**Comment by lancefisher** [comment]
I just watched the episode tonight :)

Not that anyone here needs it, but here are the steps to reproduce (on OSX):

## From the show
![image](https://user-images.githubusercontent.com/111022/33523608-7f536d0e-d7c8-11e7-9fe5-1bbe9e23ed33.png)

```bash
# Install some tools
brew install wget unrar zbar

# download the url in the email
wget https://sandbox.vflsruxm.net/plans.rar -O plans-base64.txt

# show the file for posterity
cat plans-base64.txt
UmFyIRoHAQDz4YLrCwEFBwAGAQGAgIAAaqQxujQCAwvdBwSpCSDmLtDfgAMAGGpC
b3VhcUs5UjhqWHhmcEU2a0dWLnBuZwoDAuhpGqmXR9MBykrZA0BlRWMiPzYAaWtG
4y5CKWrcagjRpWw+CVQlthXGUgSkhQFSEhBGwaFVq2EkwkYSBZ6Zla40lwY0rRaF
YNLMIiQT4VjFG+hgzEBJRKSWJUkjMwKs+JICQKJIEmY5LLcuW9n5fczubmbvebzu
83e/od3nd3m973dzeeec5mefmqzezsyKE+oJECBAomSmJ6C7VNf855mq/uDUU3vp
4Dq97Nz5F/BaAle0uC8p5xgnMS0+g+9yy3BrKldbq+D5tF18/XHvSipzd2CSz2qa
buoI+K3HqzKb8us7zREMzox1pxL3fDuZSJ/WjIHmVg5n3GB8J49zsgrIpCs3o2Fw
elehOp1zb+sPUBNrznGIbSYOKAZ9UXYnXzIr+eO7tbYlRR4HZT1ZQf4cVQpoG8PS
EZMt+g1q0Oe9/0uAZ8X05JXXez3sxGm3+uFtQK6+Hto065Ku72KlNiPKTQIeyT5l
crgkTqA4NF1QqPR7OXBmUrhWxcrzIaRqr/WBQXahXUrJtkXnJpgsd0wEboBRIbP4
sfItuGqV+KqWOblf5Ot64Hvy6jXyUEMgiDY51dmWVjyMKv5G5LdwZyyIJLKbIri1
C6Te+w/dZRNa+LP+Nt0x/ZZdyAlVWR3jJ/scC6msO/Q+4WmJ7n6Kzk80AobYK4pI
DOwx2Uei8FCZS/JFYrCthCdW4T4VpcoPaJZY3lI7Rfx6qhjsmRWpLoFK0b7aLR9l
YfKROnNI885Mp2ywOKkoSxboKOJlVp7WVGwTsTQcNxc7vKbP/yEEPbVJRb8ZdbtE
PVMYEWp6/CnHuhT7N1CWQP+lLHLU3LKw5+NWfyakyxF4hXHJAHMZc2t0UdKNrA8m
p7FsXvA/shbhioeagOCmZTIkFWPjXHwylCrUYlP8AOPrfEJur1n4og/+/HL/c6Y5
pksv3JtvDwlx6u78GsXPu7kmlFPzO1sO/p9Lt0FKp9MgDSZdvauPtO9lWFIJfHp7
/8N08CseGojeBhTMJQokVvyljS7QLP20EevozBIfJAZ+C7fFYeeaDvOmkqvrU0ip
jgIWnUv/IJbAhKH3Cr+jg5aUhW4UTHd3Q0wdnNq4afNttSRwaJuX7MGjFTdewGM9
LbYN/wIRSwib3+M2mGuywZ92YVVFpjZ4vMhNMSw9eLUFaMhfDXMnUe2M51XusaVF
qiQCyf63UDCPCQ8P0bfElu/UvS2+NyX0ALh+pSx9dxrjn3tmO9htRYwHwu5ebeXR
ZD6p9CyMpd4is9eDfl9+IspNGtGMaOntIJiBc8RzmvR4oOxlK3jP9ZJ4Rc+Qu4hx
k+O/EeVJkZ2Y6hDg/OAdd1ZRAwUEAA==%

# base64 decode the file
openssl base64 -d -in plans-base64.txt -out plans.rar

# extract the rar
unrar e plans.rar

# get the info from the qr code
zbarimg jBouaqK9R8jXxfpE6kGV.png
```

which will show:

```
QR-Code:https://github.com/RedBalloonShenanigans/MonitorDarkly
scanned 1 barcode symbols from 1 images in 0.05 seconds
```

🎉

**Comment by zinoharo** [comment]
does anyone know where the redballoon sec offices are?
i know somewhere around penn station

**Comment by cdclawson** [comment]
This is spooky. I'm finally catching up on Mr. Robot (that's what brought me here) and I'm watching it on Prime Video on my computer. The monitor I'm watching it on? A Dell Ultrasharp 2410. No wonder Elliot's monitor looked familiar...
